pubsub_logging/utils.py

```diff
@@ -49,23 +48,41 @@ def compat_urlsafe_b64encode(v):
         return base64.urlsafe_b64encode(v)
 
 
-def get_pubsub_client(http=None):  # pragma: NO COVER
-    """Return a thread local Pub/Sub client.
+def get_pubsub_client(http=None, credentials=None):
+    """Return a Pub/Sub client.
 
     Args:
       http: httplib2.Http instance. Defaults to None.
     Returns:
       Cloud Pub/Sub client.
     """
-    if not hasattr(clients, 'client'):
+    if not credentials:
         credentials = GoogleCredentials.get_application_default()
-        if credentials.create_scoped_required():
-            credentials = credentials.create_scoped(PUBSUB_SCOPES)
-        if not http:
-            http = httplib2.Http()
-        credentials.authorize(http=http)
-        clients.client = discovery.build('pubsub', 'v1beta2', http=http)
-    return clients.client
+    if credentials.create_scoped_required():
+        credentials = credentials.create_scoped(PUBSUB_SCOPES)
+    if not http:
+        http = httplib2.Http()
+    credentials.authorize(http=http)
+    return discovery.build('pubsub', 'v1beta2', http=http)
+
+
+def check_topic(client, topic, retry=3):
+    """Checks the existance of a topic of the given name.
+
+    Args:
+      client: Cloud Pub/Sub client.
+      topic: topic name that we publish the records to.
+      retry: number of retry upon intermittent failures, defaults to 3.
+
+    Returns:
+      True when it confirmed that the topic exists, and False otherwise.
+    """
+    try:
+        client.projects().topics().get(topic=topic).execute(num_retries=retry)
+        return True
+    except Exception:
+        traceback.print_exc(file=sys.stderr)
```

Comment:
Yeah, if I just pass, I'm afraid you will lose the opportunity to know the root cause (like 404, 403).